### PR TITLE
Form Page Render

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
image

**Features**
- News Feed is server-side rendered
- Form types
- Form backend and helper functions (by @Jimmy-Tempest)
- Fields render on Form page
- Form control for Form page

### CODE REVIEW

**Comment by SiravitPhokeed** [comment]
Quite bafflingly, this does build on my machine. @Jimmy-Tempest, can I see the Vercel logs?

File: src/components/news/NewsFeed.tsx [hunk 1]
```diff
@@ -50,7 +50,10 @@ const NewsFeed = ({
                 transition={animationTransition}
               >
                 <NewsCard
-                  newsItem={newsItem}
+                  // (@SiravitPhokeed)
+                  // I have no idea what’s wro here so…um…apologies
+                  // for the `any`.
+                  newsItem={newsItem as any}
```
Comment (Jimmy-Tempest) [inline src/components/news/NewsFeed.tsx:56]: any? you sure? Should we have an issue on this?

File: src/utils/types/common.ts [hunk 2]
```diff
@@ -24,6 +26,10 @@ export interface SubmittableDialogProps extends DialogProps {
   onSubmit: () => void;
 }
 
+export type BackendReturn<T, U = []> =
```
Comment (Jimmy-Tempest) [inline src/utils/types/common.ts:29]: Ahh yes, generic my favorite

File: src/components/news/NewsFeed.tsx (hunk 1 above)
Comment (SiravitPhokeed) [reply src/components/news/NewsFeed.tsx:56]: Looks like I forgot to remove some old code on converting dates. Crisis averted!

File: src/pages/s/[classNumber]/teachers.tsx [hunk 3]
```diff
@@ -202,7 +202,7 @@ export const getServerSideProps: GetServerSideProps = async ({
   params,
   req,
 }) => {
-  const redirect = await protectPageFor("admin", req);
+  const redirect = await protectPageFor("student", req);
```
Comment (Jimmy-Tempest) [inline src/pages/s/[classNumber]/teachers.tsx:205]: great

**Review by Jimmy-Tempest — APPROVED** [review]
LGTM!

File: src/pages/s/[classNumber]/teachers.tsx (hunk 3 above)
Comment (SiravitPhokeed) [reply src/pages/s/[classNumber]/teachers.tsx:205]: Yep, not sure what happened there. Probably copy-paste.

File: src/utils/types/common.ts (hunk 2 above)
Comment (SiravitPhokeed) [reply src/utils/types/common.ts:29]: I should probably ask, is this criticism or just an aside?

Comment (Jimmy-Tempest) [reply src/utils/types/common.ts:29]: just an aside

**Pull request merged**